libraries/chain/include/eosio/chain/block_header_state.hpp

```diff
@@ -110,6 +111,9 @@ struct block_header_state {
    // in the history of the blockchain so far that is not in proposed state (so either pending or active state)
    digest_type                         last_pending_finalizer_policy_digest;
 
+   // tracks the block number at which the last pending finalizer policy first became pending
```

Comment:
If no pending policy, does this use active policy's block number (to match `last_pending_finalizer_policy_digest` semantics) or the name should be changed.

If for real pending, we don't even need it as a separate field, you can find it from `pending_finalizer_policy`'s first member.

Reply:
From Areg:

> It is always the block number at which the last pending policy first was promoted to pending.
> 
> If the last pending policy is the current active policy, then it is the block number at which that active policy first was promoted to pending.
> 
> If the last pending policy is the genesis policy (and therefore it went straight to active without a pending stage), then it is the genesis block number.

Reply:
Let's make sure we add this in the comment.